packages/federation/tests/unavailable-subgraph/yoga.gateway.test.ts

```diff
@@ -0,0 +1,289 @@
+      },
+    ];
+
+    describe('EXPECTED RESPONSE - EVERYTHING OK', () => {
+      it('subgraph2_Nullable.testErrorQuery - receive valid error if testErrorQuery throw an error', async () => {
+        const expectedErrors: GraphQLFormattedError[] = [
+          {
+            message: 'My original subgraph2 error!',
+            path: ['testNestedField', 'subgraph2_Nullable', 'testErrorQuery'],
+          },
+        ];
+
+        const response = await ctx.yoga.fetch('http://yoga/graphql', {
+          method: 'POST',
+          body: JSON.stringify({ query: nullableQuery }),
+          headers: { 'content-type': 'application/json' },
+        });
+
+        const result: FormattedExecutionResult = await response.json();
+
+        expect(response.status).toBe(200);
+        expect(result.data).toMatchObject(expectedNullableData);
+        expect(result.errors).toMatchObject(expectedErrors);
+      });
+
+      it('subgraph2_NonNullable.testErrorQuery - receive valid error if testErrorQuery throw an error', async () => {
+        const expectedErrors: GraphQLFormattedError[] = [
+          {
+            message: 'My original subgraph2 error!',
+            path: [
+              'testNestedField',
+              'subgraph2_NonNullable',
+              'testErrorQuery',
+            ],
+          },
+        ];
+
+        const response = await ctx.yoga.fetch('http://yoga/graphql', {
+          method: 'POST',
+          body: JSON.stringify({ query: nonNullableQuery }),
+          headers: { 'content-type': 'application/json' },
+        });
+
+        const result: FormattedExecutionResult = await response.json();
+
+        expect(response.status).toBe(200);
+        expect(result.data).toMatchObject(expectedNonNullableData);
+        expect(result.errors).toMatchObject(expectedErrors);
+      });
+    });
+
+    describe('UNEXPECTED RESPONSE - FAILING TESTS', () => {
+      beforeAll(async () => {
+        await ctx.subgraph2.stop();
+      });
+
+      afterAll(async () => {
+        await ctx.subgraph2.start();
+      });
+
+      it('subgraph2_Nullable.testErrorQuery - subgraph2 is unavailable', async () => {
+        // it should be "expectedNullableData"
+        const unexpectedData = {
+          testNestedField: {
+            ...expectedNullableData.testNestedField,
+            subgraph2_Nullable: null,
+          },
+        };
+
+        // the "ECONNREFUSED" error got completely lost somewhere along the way !!!
+        const unexpectedErrors = undefined; // it should be "expectedUnavailableErrors"
+
+        const response = await ctx.yoga.fetch('http://yoga/graphql', {
+          method: 'POST',
+          body: JSON.stringify({ query: nullableQuery }),
+          headers: { 'content-type': 'application/json' },
+        });
+
+        const result: FormattedExecutionResult = await response.json();
+
+        expect(response.status).toBe(200);
+        expect(result.data).toMatchObject(unexpectedData);
```

Comment:
@ardatan This is interesting. If this test runs on Bun, it correctly receive `expectedNullableData` mentioned above. 🤔